homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -93,14 +97,174 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
     event_handler: &mut EventHandler<DB>,
 ) {
     match event {
-        // TODO: add identify for adding compatable kademlia nodes.
         // TODO: use kademlia to discover new gossip nodes.
+        SwarmEvent::Behaviour(ComposedEvent::Identify(identify_event)) => {
+            match identify_event {
+                identify::Event::Error { peer_id, error } => {
+                    error!(err=?error, peer_id=peer_id.to_string(), "error while attempting to identify the remote")
+                }
+                identify::Event::Sent { peer_id } => {
+                    debug!(peer_id = peer_id.to_string(), "sent identify info to peer")
+                }
+                identify::Event::Received { peer_id, info } => {
+                    debug!(peer_id=peer_id.to_string(), info=?info, "identify info received from peer");
+
+                    if !event_handler
```

Comment:
minor: may be nice to chain this via [bool::then_some](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some) or similar.

Reply:
I would prefer to keep as-is. it is more readable than a long chain wrapped with paren, e.g. 
```
(!long.chain(|| {})
    .of()
    .events())
    .then(|| {})
```

Reply:
Yeah. That's fine for now. I think there's room for some general cleanup in this section, but we can refactor later for sure.